[29754] Don't match substates in the /boards state definition

oliverguenther opened the pull request:
The trailing slash of the board route ensures that it is not matched in routes such as `/projects/boards/foo`.

https://community.openproject.com/wp/29754

oliverguenther committed:
Fix board url matching project name

HDinger commented:
@oliverguenther the error still occurs for me. However there is no error message on the console any more. If you can't make it today, I'll take care of it tomorrow.

<img width="1039" alt="Bildschirmfoto 2019-03-20 um 16 09 46" src="https://user-images.githubusercontent.com/7457313/54695635-9ca0bc00-4b2a-11e9-8254-9ebff6da5a35.png">

oliverguenther committed:
Treat board_id as number

Otherwise it will parse `work_packages/1234/activity` as the ID and
match the board state
